lib/canary/plugs.ex

```diff
@@ -222,21 +222,28 @@ defmodule Canary.Plugs do
   defp fetch_resource(conn, opts) do
     repo = Application.get_env(:canary, :repo)
 
+    id = case opts[:id] do
+      nil ->
+        conn.params["id"]
+      resource_id ->
+        conn.params[resource_id]
+    end
```

Comment:
This is a really minor thing, but I think this block could be moved into a small function e.g. 
```
id = get_resource_id(opts)

defp get_resource_id(opts) do
  case opts[:id] do
    nil ->
      conn.params["id"]
    resource_id ->
      conn.params[resource_id]
    end
end
```